tests/unit/transaction_queue.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "interpreter_faker.hpp"
+
+#include <gtest/gtest.h>
+#include <chrono>
+#include <stop_token>
+#include <string>
+#include <thread>
+#include "gmock/gmock.h"
+
+/*
+Tests rely on the fact that interpreters are sequentially added to runninng_interpreters to get transaction_id of its
+corresponding interpreter/.
+*/
+class TransactionQueueSimpleTest : public ::testing::Test {
+ protected:
+  memgraph::storage::Storage db_;
+  std::filesystem::path data_directory{std::filesystem::temp_directory_path() / "MG_tests_unit_transaction_queue_intr"};
+  memgraph::query::InterpreterContext interpreter_context{&db_, {}, data_directory};
+  InterpreterFaker running_interpreter{&interpreter_context}, main_interpreter{&interpreter_context};
+};
+
+// This test relies on explicit transations + explicit thread killing for testing if show transactions work as expected
+TEST_F(TransactionQueueSimpleTest, ShowTransactions) {
+  std::jthread running_thread = std::jthread(
+      [this](int thread_index) {
+        running_interpreter.Interpret("BEGIN");
+        for (int i = 0; i < 5; i++) {
+          running_interpreter.Interpret("CREATE (:Person {prop: " + std::to_string(thread_index) + "})");
+          std::this_thread::sleep_for(std::chrono::milliseconds(200));
+        }
+        running_interpreter.Interpret("ROLLBACK");
+      },
+      0);
+
+  {
+    main_interpreter.Interpret("CREATE (:Person {prop: 1})");
+    auto show_stream = main_interpreter.Interpret("SHOW TRANSACTIONS");
+    ASSERT_EQ(show_stream.GetResults().size(), 2U);
+    // admin executing the transaction
+    EXPECT_EQ(show_stream.GetResults()[0][0].ValueString(), "");
+    ASSERT_TRUE(show_stream.GetResults()[0][1].IsString());
+    EXPECT_EQ(show_stream.GetResults()[0][2].ValueList().at(0).ValueString(), "SHOW TRANSACTIONS");
+    // Also admin executing
+    EXPECT_EQ(show_stream.GetResults()[1][0].ValueString(), "");
+    ASSERT_TRUE(show_stream.GetResults()[1][1].IsString());
+    EXPECT_EQ(show_stream.GetResults()[1][2].ValueList().at(0).ValueString(), "EXPLICIT TRANSACTION");
```

Comment:
Is there any smarter way to check these types and values without doing referencing of list

Reply:
Return result is list of strings so I don't think there is. What is wrong with this? If something is wrong the test will fail

Reply:
Nothing is wrong but seems kind of messy with checking this way.

Reply:
But I don't have any idea how to do it better.